scipy/optimize/tests/test__basinhopping.py

```diff
@@ -301,6 +301,27 @@ def callback2(x, f, accepted):
                      niter=10, callback=callback2, seed=10)
         assert_equal(np.array(f_1), np.array(f_2))
 
+    def test_random_gen(self):
+        # check that np.random.Generator can be used (numpy >= 1.17)
+        try:
+            # obtain a np.random.Generator object
+            rng = np.random.default_rng(1)
+        except AttributeError:
+            # only available in numpy >= 1.17
+            return
+
+        minimizer_kwargs = {"method": "L-BFGS-B", "jac": True}
+
+        res1 = basinhopping(func2d, [1.0, 1.0],
+                            minimizer_kwargs=minimizer_kwargs,
+                            niter=10, seed=rng)
+
+        rng = np.random.default_rng(1)
```

Comment:
Line 319 here seems the same as the line 308 up in try/catch block?
Is the intent of line 308 to test if the version of numpy >= 1.17 is available in the env?

Reply:
This one can be a skipif.